truck_inspection_client/src/truck_inspection_client.cpp

```diff
@@ -79,6 +80,30 @@ namespace truck_inspection_client
     void TruckInspectionClient::requestCallback(const cav_msgs::MobilityRequestConstPtr& msg)
     {
         if(msg->strategy == this->INSPECTION_STRATEGY) {
+            std::uint16_t vin_retrive_count = 0;
+            //Checking: vin number does not exist yet in global parameter server
+            while((!pnh_->getParam("/vin_number", vin_number_)) && vin_retrive_count < MAX_RETRIEVE_VIN_COUNT)
```

Comment:
This loop should be located in the initialize() method. Not hear. If the parameter is not loaded by this point it never will be.

Reply:
Addressed: https://github.com/usdot-fhwa-stol/carma-platform/pull/1020/commits/56dd79335ca2255d024912c27d7e13719a88414d